test/OpenTelemetry.Instrumentation.AspNetCore.Tests/MetricTests.cs

```diff
@@ -0,0 +1,136 @@
+// <copyright file="MetricTests.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc.Testing;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Tests;
+using OpenTelemetry.Trace;
+#if NETCOREAPP2_1
+using TestApp.AspNetCore._2._1;
+#elif NETCOREAPP3_1
+using TestApp.AspNetCore._3._1;
+#else
+using TestApp.AspNetCore._5._0;
+#endif
+using Xunit;
+
+namespace OpenTelemetry.Instrumentation.AspNetCore.Tests
+{
+    public class MetricTests
+        : IClassFixture<WebApplicationFactory<Startup>>, IDisposable
+    {
+        private readonly WebApplicationFactory<Startup> factory;
+        private MeterProvider meterProvider = null;
+
+        public MetricTests(WebApplicationFactory<Startup> factory)
+        {
+            this.factory = factory;
+        }
+
+        [Fact]
+        public void AddAspNetCoreInstrumentation_BadArgs()
+        {
+            MeterProviderBuilder builder = null;
+            Assert.Throws<ArgumentNullException>(() => builder.AddAspNetCoreInstrumentation());
+        }
+
+        [Fact]
+        public async Task RequestMetricIsCaptured()
+        {
+            var metricItems = new List<MetricItem>();
+            var metricExporter = new TestExporter<MetricItem>(ProcessExport);
+
+            void ProcessExport(Batch<MetricItem> batch)
+            {
+                foreach (var metricItem in batch)
+                {
+                    // The exporter is invoked even when there are no metrics
```

Comment:
Would it make sense to not invoke the exporter in a pipeline when there's nothing to export?

Reply:
For logs and traces we do the same (when there is nothing in the batch, we don't call exporter) https://github.com/open-telemetry/opentelemetry-dotnet/blob/8a087d50d6f19980bc192ea735f7099c34d95d78/src/OpenTelemetry/BatchExportProcessor.cs#L218.

For metrics I am not sure, maybe we can start with it (don't call exporter when there is nothing), and we can change it later if there is a need - e.g. some exporters might want to send heart beats every 5 secs even if there is no data.

Reply:
Opened #2191 for discussion